### DISCUSSION TITLE
v17.2

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1906] [Announcements]
This is an ambitious update which will hopefully transform your mosh experience for some of you. Say hello to the self-installing mosh server. Now, Blink will seamlessly upgrade your connection during mosh bootstrap. To do this and laying the foundation of more stuff to come, ve've also reimplemented the Mosh client on Swift. To top it off, we've fine-tuned SSH config file parsing and squashed pesky bugs, and we did a lot of other housekeeping.

This update touches key functionality in Blink, so we need your help to put these enhancements to the test! Dive into the details below, and let us know your thoughts and findings.

### Self-Installing Mosh Server
- Blink can now automatically upgrade connections from SSH to Mosh during Mosh bootstrap!
- **Test Needed**: Help us ensure smooth functionality across various architectures and platforms.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

### Reimplemented Mosh Client on Swift
- Key for upcoming features and enables self-installation.
- We keep the old mosh under `mosh2` command in case of issues.
- **Test Needed**: Verify the CLI functions as expected.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

### SSH Config File Parsing Improvements
- Enhanced parsing for SSH config files.
- **Test Needed**: Ensure ssh config still meet your expectations.
- 🔗 [Issue 1809](https://github.com/blinksh/blink/issues/1809)

### SSH Password Prompt Information
- Improved information on SSH password prompts.
- **Test Needed**: Check the new prompts and let us know what you think.
- 🔗 [Issue 1902](https://github.com/blinksh/blink/issues/1902)

### Auto-Lock with Configurable Timeout Options
- Introducing auto-lock with configurable timeout options.
- **Customization**: Configure timings to suit your preferences.
- 🔗 [Issue 1903](https://github.com/blinksh/blink/issues/1903)

### New Key Creation Prompt
- Access a new key creation prompt in settings.
- **Feedback needed**: Let us know what you think of the copy.
- 🔗 [Issue 1902](https://github.com/blinksh/blink/issues/1902)

### Improved Host Creation Consistency
- Resolved host disappearance after creation, ensuring eventual consistency.
- **Test Needed**: Help test iCloud sync; report any anomalies.
- 🔗 [Issue 1904](https://github.com/blinksh/blink/issues/1904)

### SSH Args Parsing Fix
- Fixed parsing issue with SSH arguments at the end.
- **Test Needed**: Validate parsing meets your expectations.
- 🔗 [Issue 1909](https://github.com/blinksh/blink/issues/1909)

### Config Sync with ForwardAgent Flag
- Config now synchronizes ForwardAgent flag with SSH config.
- 🔗 [Issue 1909](https://github.com/blinksh/blink/issues/1909)

---

If you encounter any issues or have suggestions, please share them in the respective GitHub issues linked above. For general comments, please drop them here!

**Comment by carloscabanero** [comment]
## Build 855

### Updated LibSSH to 0.9.8+
- Resolved Terrapin issues and other security patches.
- 🔗 [Issue 1918](https://github.com/blinksh/blink/issues/1918)

### New Mosh Client Enhancements
- Mosh auto-install now identifies Aarch64 architecture.
- New Mosh client includes all flags from the previous client.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

### Fixes for New Mosh client
- Resolved issue causing New Mosh to block on exit when the app had been in the background after restore.
- Fixed app crash issue in New Mosh due to an early exit, throwing a kill in the wrong place when the host name could not be resolved.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

**Comment by carloscabanero** [comment]
## Build 861

### Pragmata Pro Update
- Pragmata Pro updated to version 830, now featuring Nerd fonts, ligature tweaks, and improved characters.
- 🔗 [Issue 1946](https://github.com/blinksh/blink/issues/1946)

### Enhanced Mosh Functionality
- Mosh now supports Happy Eyeballs on "default" IP resolution mode.
- Introducing `--6` and `--4` flags to force connections over IPv6 or IPv4 when necessary.
- 🔗 [Issue 269](https://github.com/blinksh/blink/issues/269)
  - **Note**: Please refer to the issue for additional information.

### Mosh Auto-Install Enhancement
- Mosh will now auto-install with the `--install-static` flag. This was a concern of some users with high-security constraints, avoiding UI-triggered installations on the remote by mistake.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

### Smart Keys Click Sounds Fix
- Resolved issue with Smart Keys click sounds playing at very loud volumes.
- 🔗 [Issue 1937](https://github.com/blinksh/blink/issues/1937)

### Port Forwarding Improvement
- Port forwarding now supports the `GatewayPorts` flag.
- When set to "yes," ports are exposed to other machines in the network.
- 🔗 [Issue 1915](https://github.com/blinksh/blink/issues/1915)

### External Display Alert Fix
- Alerts on the External Display in mirror mode issue resolved.
- No more alerts showing on the external display.
- 🔗 [Issue 1944](https://github.com/blinksh/blink/issues/1944)

### Files.app stability fixes
- As part of the LibSSH update, we have fixed a couple bugs in our SFTP subsystem that could provoke a race condition that would block the extension.

### iOS Stability Enhancements
- Resolved New iOS crashes.
- 🔗 [Issue 1945](https://github.com/blinksh/blink/issues/1945)

**Comment by carloscabanero** [comment]
## Build 863

### Previous Mosh renamed
- The previous mosh can now be found under the `mosh1` command.
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1905)

**Comment by carloscabanero** [comment]
## 16.2.1 Build 865

### Clipboard fixes
- Mosh sends spurious OSC52 on Blink restart
- 🔗 [Issue 1905](https://github.com/blinksh/blink/issues/1956)

### New Keys UI fixes
- The new Keys UI was not linking properly to readme and some actions.

**Comment by iflowerml** [comment]
### Mosh-server binary fix 

- There is a known issue issue with installing mosh-server on some Linux machines. A fix is coming soon! 
- 🔗 [Issue 1963](https://github.com/blinksh/blink/issues/1963)
